crates/red_knot_python_semantic/resources/mdtest/directives/assert_type.md

```diff
@@ -0,0 +1,123 @@
+from typing import reveal_type
+
+# `assert_type`
+
+## Basic
+
+```py
+from typing_extensions import assert_type
+
+def _(x: int):
+    assert_type(x, int)  # fine
+    assert_type(x, str)  # error: [type-assertion-failure]
+```
+
+## Narrowing
+
+The asserted type is checked against the inferred type, not the declared type.
+
+```py
+from typing_extensions import assert_type
+
+def _(x: int | str):
+    if isinstance(x, int):
+        reveal_type(x)  # revealed: int
+        assert_type(x, int)  # fine
+```
+
+## Equivalence
+
+The actual type must match the asserted type precisely.
```

Comment:
Again, we should specify somewhere what "precisely" or "exactly the same" means. Because it's not "structural equivalence" or `Eq` on `Type` (because `int | str` and `str | int` should match, for example)